x-pack/plugins/security_solution/public/common/containers/source/index.tsx

```diff
@@ -78,29 +78,47 @@ interface FetchIndexReturn {
   browserFields: BrowserFields;
   indexes: string[];
   indexExists: boolean;
+  /**
+   * @deprecated use dataViewSpec
+   */
   indexPatterns: DataViewBase;
+  /**
+   * This variable should be renamed dataViewSpec and should have it's types
+   * updated upstream but we will save that for some other time
+   */
   dataView: DataViewSpec | undefined;
+  dataViewSpec: DataViewSpec | undefined;
 }
 
 /**
  * Independent index fields hook/request
  * returns state directly, no redux
  */
 export const useFetchIndex = (
-  indexNames: string[],
+  indexNamesOrDvId: string | string[],
```

Comment:
If we're trying to move away from the `indexNames` pattern, would it be better to just construct a new param structure and separate these two arguments instead of delineating on `isArray`? I understand that would cause a lot of refactor since we use this hook everywhere, but just wondering what your thoughts were. Having multiple variables in here with "or" in the name reads a bit confusing